src/tests/core/engine/app_engine_processor_test.py

```diff
@@ -0,0 +1,25 @@
+import unittest
+from unittest.mock import patch
+
+from src.core.engine.app_engine_processor import AppEngineProcessor
+from src.core.engine.processors.kitchen_simulator import KitchenSimulator
+
+
+class TestAppEngineProcessor(unittest.TestCase):
+    @patch("src.core.engine.app_engine_processor.AppProcessorConfig")
+    @patch("src.core.engine.app_engine_processor.KitchenSimulator")
+    def test_mock_in_app_context(self, mock_kitchen, mock_config):
```

Comment:
Make `mock_config` to return a `build_app_engine_config` when the instance is created. Remove the mock for KitchenSimulator.

Then assert that `len(app_engine_processor.app_context.processors) = 1` and `app_engine_processor.app_context.processors[0].app_processor_config.id` is equal to the id youi used here `build_app_engine_config`

Reply:
Done.